tests/nixpkgs-module.nix

```diff
@@ -28,7 +28,54 @@ let
 in
 linkFarmFromDrvs "nixpkgs-module-test" [
 
-  # TODO: expect not setting `nixpkgs.pkgs` to throw
+  # Test that pkgs-config is affected by `nixpkgs.config`
+  (testModule "nixpkgs-config" (
+    { pkgs, ... }:
+    {
+      nixpkgs.config = {
+        permittedInsecurePackages = [
+          "foobar123"
+        ];
+      };
+
+      nixpkgs.hostPlatform = {
+        inherit (stdenv.hostPlatform) system;
+      };
+
+      assertions = [
+        {
+          assertion = pkgs.config.permittedInsecurePackages == [ "foobar123" ];
```

Comment:
There would be no added value to try to actually install an insecure package right? If we observe the config value it should mean that the package would be installed?

Reply:
Probably not. All this stuff is well tested in upstream nixpkgs too, so our tests are only really sanity checks to ensure the module isn't massively broken.

I've tried to make most of them lightweight unit tests rather than realistic integration tests.